### PR TITLE
specify UpdateStrategy (RollingUpdate) and AntiAffinity(Hostname) for ElasticSearch data pods

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Specify UpdateStrategy (RollingUpdate) and AntiAffinity(Hostname) for ElasticSearch data pods

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Review by xrstf — APPROVED** [review]
LGTM

File: config/logging/elasticsearch/templates/es-data-stateful.yaml [hunk 1]
```diff
@@ -14,6 +16,16 @@ spec:
         component: elasticsearch
         role: data
     spec:
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - podAffinityTerm:
```
Comment (alvaroaleman) [inline config/logging/elasticsearch/templates/es-data-stateful.yaml:22]: Shouldn't this be anti-affinity? Or do you want them on the same host to speed up replication?

Comment (mrIncompetent) [reply config/logging/elasticsearch/templates/es-data-stateful.yaml:22]: The decision if using `affinity` or `aniti-affinity` happens 2 levels above via `affinity.podAntiAffinity`, `affinity.podAffinity` or `affinity.nodeAffinity`.

`preferredDuringSchedulingIgnoredDuringExecution.podAffinityTerm` defines just defines the "where to apply it".

Comment (mrIncompetent) [reply config/logging/elasticsearch/templates/es-data-stateful.yaml:22]: And: We use exactly that in the etcd StatefulSet. All etcd pods get scheduled on different nodes (if possible)

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**